src/ripple/protocol/STObject.h

```diff
@@ -401,7 +401,7 @@ class STObject
         if (!cf)
             throw std::runtime_error ("Wrong field type");
 
-        cf->setValue (value);
+        cf->setValue (std::move (value));
```

Comment:
I had not considered the case where V was explicitly set to be a reference type. I will add a static_assert, as Scott suggests. Thanks Scott & Howard for pointing out a case I had not considered.